### ISSUE TITLE
Calculation of blazed number μ

### ISSUE THREAD

**Issue by NiuYihan1999** [wavefrontshaping/tutorial-DMD-setup-2023#2]
Thank you for your great work, it helps a lot.

I tried to derive the equation of the blazed number μ by myself. And I found that I can't understand why 2D incident angle α' can be projected to 1D. So I changed another way by projecting pixel pitch, and got another equation: μ = |d / √2 / λ × (sin(α') + sin(θ')), mod 1|, where the α' is the original incident angle, and θ' = 2θ_DMD - α' is the original outgoing angle, that is, all are same as your definations.

Could you please tell me if my equation is right?

P.S. In your previous works ([this](https://www.wavefrontshaping.net/post/id/21) and [this](https://www.wavefrontshaping.net/post/id/86)), there is [a comment](https://www.wavefrontshaping.net/post/id/21#comment-6350370424) , it also raised the same problem, and our equations are actually the same.

**Comment by wavefrontshaping** [comment]
Hi,

Sorry, I did make mistakes on this calculation, It has already been brought to my attention very recently and I am about to correct the paper as soon as I can.

- Indeed, the first mistake I made was to project to 1D the component of θ' = 2θ_DMD - α' instead of the resulting angle. 
- Then, and this one is still not clear in my mind, I used the projection formula a_1D = arctan(tan(a)/sqrt(2)), which I thought was well established and I did originally find this one writing the equations. However, if you consider angles as phase slopes e^sin(a)*x,  the projection gives a_1D = arcsin(sin(a)/sqrt(2))

Adding those corrections led to the much simpler formula you found, but it is still not clear in my mind why the tan() formula does not work. 
Note that in both cases, for the small angles, all formulas will give approximately the same thing, probably the reason I missed the errors in the first place...

**Comment by NiuYihan1999** [comment]
Thank you for your response!

Actually I'm still a little bit confused about this: α_1D ＝ asin(sin(α') / √2). I can't really find out its physical significance.

On the other hand, when I just plug this α_1D and θ_1D into your original equation: μ = |d / λ × (sin(α_1D) + sin(θ_1D)), mod 1|, where θ_1D ＝ 2 × asin(sin(θ_DMD) / √2) - α_1D. It doesn't match my equation: μ = |d / √2 / λ × (sin(α') + sin(θ')), mod 1|. But if you just change θ_1D into: θ_1D ＝ asin(sin(2 × θ_DMD - α') / √2), it will match my equation.

Like I said. I still don't know the physical significance of all equations except mine. While equations above are indeed the mathematical results.

**Comment by wavefrontshaping** [comment]
Agreed, I was not clear, that is exactly what I meant.
My first point was that in 2 × θ_DMD - α', I initially converted each angle (θ_DMD and α') to 1D and then applied the formula, which is wrong.
If you convert instead after computing 2 × θ_DMD - α, you have a_1D = arcsin(sin(a)/sqrt(2)) with a =  2 × θ_DMD - α', thus the formula you found.
Do you agree?

**Comment by NiuYihan1999** [comment]
Yes, I totally agree that mathematically. Though I don't know its physical significance.

By the way, I found my equation: μ = |d / √2 / λ × (sin(α') + sin(θ')), mod 1| through another way. That is, never project any angle, just consider the optical path difference of two adjacent micromirrors, which is: d / √2 × (sin(α') + sin(θ')), where d / √2 is the distance of those two adjacent micromirrors projected to real horizontal plane. Then just divide that optical path difference by λ. And in your way, you project α' and θ' into α_1D = arcsin(sin(α') / √2) and θ_1D ＝ arcsin(sin(2 × θ_DMD - α') / √2). Therefore, your equation will be mathematically equivalent to mine.

Below is the diagram of how to calculate d / √2 based on your figure:
![e1131251c33f7ccbd4d54f53aecc8e6](https://github.com/user-attachments/assets/6150194b-3803-4bcd-a114-f969ef045dae)

**Comment by wavefrontshaping** [comment]
It seems to be equivalent, but it was not obvious to me initially. Indeed, the grating you have in the 45-degree angle, so with the d/sqrt(2) period in the axis we can call x', has a slanted unit cell, the repeated pattern. I was not initially convinced we can treat it similarly to a "standard" grating.

**Comment by NiuYihan1999** [comment]
Yes, it's a little bit hard to think. But I just imagine all the reflected outgoing light as parallel light with a special phase modulation.

Same amplitude and propagation direction with phase like this: 
![image](https://github.com/user-attachments/assets/450365af-888f-4b43-933d-d7c650678ee4)

**Comment by wavefrontshaping** [comment]
Hi,

I agree with the final results but the justification was not obvious. 
Indeed, on your first picture of the DMD pixels, there is no periodicity in the plane of the incident and output beams with d/sqrt(2): if you translate the whole array by this amount in the horizontal direction, you do not end up with the same array.

My mistake was to try to project thins into this axis where there is no d-periodicity, while I should have done the opposite and express the effect of the 45 degrees mirror tilt into the axes of the array.
I will update everything soon, and will also retract and resubmit the paper.

Thank you for your help!
